src/workerd/api/kv-test.wd-test

```diff
@@ -10,8 +10,18 @@ const unitTests :Workerd.Config = (
         bindings = [ ( name = "KV", kvNamespace = "kv-test" ), ],
         compatibilityDate = "2023-07-24",
         # "experimental" flag is needed to test deleteBulk
-        compatibilityFlags = ["experimental", "nodejs_compat", "service_binding_extra_handlers"],
+        compatibilityFlags = ["experimental", "nodejs_compat", "service_binding_extra_handlers", "streaming_tail_worker", "tail_worker_user_spans"],
+        streamingTails = ["tail"],
       )
     ),
+    (name = "tail", worker = .tailWorker, ),
   ],
 );
+
+const tailWorker :Workerd.Worker = (
+  modules = [
+    (name = "worker", esModule = embed "kv-instrumentation-test.js")
+  ],
+  compatibilityDate = "2024-10-14",
+  compatibilityFlags = ["experimental", "nodejs_compat", "streaming_tail_worker"],
```

Comment:
Nit: Currently the STW compat flag is only needed on tailed workers, dropping it here would make that more clear (of course we're aiming to do the opposite but that'll take time).